lib/internal/errors.js

```diff
@@ -120,8 +112,13 @@ const prepareStackTrace = (globalThis, error, trace) => {
   //     at function (file)
   //     at file
   let errorString;
-  if (kIsNodeError in error) {
-    errorString = `${error.name} [${error.code}]: ${error.message}`;
+  // Do not use primordials here: we intercept user code here.
+  if (typeof error.toString === 'function' &&
+      error.toString !== MainContextObjectToString) {
+    errorString = error.toString();
+    if (errorString === '[Object object]') {
+      errorString = ErrorPrototypeToString(error);
+    }
```

Comment:
We should take into account the `toString` may be a getter that returns a different value (or throw) on subsequent calls.
```suggestion
  const errorToString = error.toString;
  if (typeof errorToString === 'function' &&
      errorToString !== MainContextObjectToString) {
    errorString = FunctionPrototypeCall(errorToString);
    if (errorString === '[Object object]') {
      errorString = ErrorPrototypeToString(error);
    }
```

Reply:
I already thought about it and I tested it as well: if a new error is thrown during the prepare step, that error is going to take over and is reported instead. That should exactly be the behavior that we want in this case. As such, I went for the simple solution.

Reply:
What about the following case:

```js
const error = new Error;
let hasAlreadyBeenCalled = false;
Reflect.defineProperty(error, 'toString', {
  __proto__: null,
  get() {
    if (!hasAlreadyBeenCalled) {
      hasAlreadyBeenCalled = true;
      return () => {};
    }
  },
});
throw error;
```

The first check checks one value, the other checks check another value, resulting in broken code. By desctructuring the value, you remove that footgun.

Reply:
I added your suggestion.